esp-hal-embassy/src/lib.rs

```diff
@@ -47,7 +57,83 @@ use self::time_driver::{EmbassyTimer, Timer};
 mod executor;
 mod time_driver;
 
-/// Initialize embassy
-pub fn init(clocks: &Clocks, time_driver: &'static mut [Timer]) {
-    EmbassyTimer::init(clocks, time_driver)
+macro_rules! mk_static {
+    ($t:ty,$val:expr) => {{
+        static STATIC_CELL: static_cell::StaticCell<$t> = static_cell::StaticCell::new();
+        #[deny(unused_attributes)]
+        let x = STATIC_CELL.uninit().write(($val));
+        x
+    }};
+}
+
+/// A trait to allow better UX for initializing the timers.
+pub trait TimerCollection {
```

Comment:
Could this be `Into`/`From` instead? (Might be missing something)

Reply:
I think so, I'll look into it, thanks.

Reply:
Unfortunately, because of the inner static variables, these traits are single-use. Calling the conversion on two different array types, for example, would end up panicking. We can document this if we use some ad-hoc traits like currently, but from a `From` impl I think this would be very surprising.